frontend/src/components/SyncButton.vue

```diff
@@ -0,0 +1,111 @@
+<template>
+  <div
+    v-if="!isLogin && unsyncedDrives.length"
+    class="col-2 button-container"
+  >
+    <div v-if="inDrivesView">
+      <div v-if="syncStatus === 'untouched'">
+        <svg
+          class="icon"
+          @click="triggerSync"
+        >
+          <use xlink:href="../assets/sprite.svg#sync-alt" />
+        </svg>
+      </div>
+      <div v-else-if="syncStatus === 'loading'">
+        <div class="spinner-border icon" />
+      </div>
+      <div v-else-if="syncStatus === 'success'">
+        <svg class="icon success">
+          <use xlink:href="../assets/sprite.svg#success" />
+        </svg>
+      </div>
+      <div v-else-if="syncStatus === 'fail'">
+        <svg class="icon fail">
+          <use
+            xlink:href="../assets/sprite.svg#fail"
+            href="../assets/sprite.svg#fail"
+          />
+        </svg>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { IS_ONLINE, SYNC, UNSYNCHRONISED_DRIVES } from '@/store/constants';
+import { mapActions, mapState, mapGetters } from 'vuex';
+import { driveListRoute, loginRoute } from '@/router/routes';
+
+export default {
+  name: 'SyncButton',
+  computed: {
+    ...mapState([IS_ONLINE]),
+    ...mapGetters({ unsyncedDrives: UNSYNCHRONISED_DRIVES }),
+  },
+  data() {
+    return {
+      isLogin: this.$router.currentRoute.path === loginRoute.path,
+      inDrivesView: null,
+      syncStatus: 'untouched',
```

Comment:
another nitpick: I guess it would be nice to use some kind of enum (or just a dict) for keeping these statuses in one place and to avoid any possible typo, but this approach works as well.

Reply:
I understand the point. I think pure numbers without meaning could be difficult to read. The dictionary idea sounds good by I don't know how this would work. I tried sth like:
`      syncStatus: {
        1: 'untouched',
        2: 'loading',
        3: 'success',
        4: 'failure',
      },`
But then v-if=syncStatus.1 was no go. It seems you cannot ascribe sheer numbers as the name of the var. 

If you got any ideas about execution I'd gladly take a suggestion. Otherwise I have no clue how to make that dict happen :P

Reply:
Well, to be fully correct - javascript does not support enums afaik, but you can simulate it with dict 
(example: https://stijndewitt.com/2014/01/26/enums-in-javascript/)

which turns out to your approach, but reversed:

    syncStatuses: { untouched: 1, loading: 2, success: 3, failure: 4 }

which lets you do things like:

    <div v-if="syncStatus === syncStatuses.untouched">

It's not really bulletproof, because you still can make up the states (and do typos) like: `... === syncStatuses.faiure`
but I suppose that will be less likely due to editor autocompletion of static keys.

Anyway, that's what I had in mind. The main benefit from my perspective is that by creating some king of map with these keys you're making it very obvious as to what states are actually there. If you'd use strings then it can get more difficult to track, because the possible statuses are scattered in the codebase.

Reply:
Ok it works :)